JASP-Engine/JASP/R/LSbinomialestimation.R

```diff
@@ -233,12 +234,14 @@ LSbinomialestimation   <- function(jaspResults, dataset, options, state = NULL){
   
   if (is.null(containerPlots[[paste0("plots",type)]])){
     
-    plotsSimple <- createJaspPlot(width = 530, height = 400, aspectRatio = 0.7)
+    plotsSimple <- createJaspPlot(
+      width  = if (options[[ifelse(type == "Prior", "plotsPriorType", "plotsPosteriorType")]] == "overlying") 700 else 530,
```

Comment:
I'm not sure what do you mean by that. Can you elaborate please?

Reply:
Sure, you have:

```r
   plotsSimple <- createJaspPlot(
      width  = if (options[[ifelse(type == "Prior", "plotsPriorType", "plotsPosteriorType")]] == "overlying") 700 else 530,
      height = 400)

    plotsSimple$position <- 2
    plotsSimple$dependOn(c(
       .dataDependenciesBinomialLS,
       ifelse(options[[ifelse(type == "Prior", "plotsPriorType", "plotsPosteriorType")]] == "overlying", "colorPalette", ""))) 
```
the same `if` statement with a nested `ifelse` appears twice. You could write:
```r
if (options[[ifelse(type == "Prior", "plotsPriorType", "plotsPosteriorType")]] == "overlying") {
  width <- 700
  dependency <- "colorPalette"
} else {
  width <- 530
  dependency <- NULL
}
```
which is more lines of code but easier to read.

Reply:
hmm, the thing is that I'm using this structure thru the whole analysis to make the dispatching somewhat concise. If I were to rewrite this one occurrence, I would have to change another 700 lines and double them. Not sure whether that improves the readability overall

Reply:
Well, to be honest, I don't find nested `ifelse` (or `ifelse` within `if`) very readable. I had to read the same line of code multiple times before I properly understood it. I particularly pointed it out here because you're repeating the same code twice in a row.

If it's always the same `ifelse(type == "Prior", "plotsPriorType", "plotsPosteriorType")`, you could also write a function for it, e.g., 
```r
getPlotsType <- function(type) if (type == "Prior") "plotsPriorType" else "plotsPosteriorType"
```
that would also reduce the chances of any typos.

Reply:
the problem is that there is around 20-40 of these variations:

`ifelse(type == "Prior", "x", "z")`

some of them used only once, some more often and it doesn't make sense to create a dedicated function. I have been playing around with an idea to use a paste function that adds either Prior or Posterior to the name of the setting, but that would require more qml rewriting and possibly breaking more things

Reply:
If the structure is always like that, you could use a static table that you index by name:

```r
table <- matrix(c(
  "plotsPriorType", "plotsPosteriorType"
), ncol = 2, 
  dimnames <- list(
    c("plotsType"), 
    c("prior", "posterior")
))
table["plotsType", type] # to get information
```
This could also be a list, if the elements have different types. Then you'd get `table[["plotsType"]][type]`.

Sticking to a convention with either "prior" or "posterior" after a setting is also fine.

I agree that if there are around 20-40 of these variations, it's better to solve this for all of them. Can you create an issue for that on internal so that we don't forget about it? You can just link to the discussion here. Then I'd just leave this code as-is for now.

Reply:
ok, done